lib/rack/proxy.rb

```diff
@@ -39,7 +39,8 @@ def reconstruct_header_name(name)
     end
 
     # @option opts [String, URI::HTTP] :backend Backend host to proxy requests to
-    def initialize(opts = {})
+    def initialize(app = nil, opts= {})
```

Comment:
This would have to be backward compatible: if the first arg is a hash, then it should be assigned to the `opts` variable.